resources/dune

```diff
@@ -0,0 +1,4 @@
+(library
+ (name xapi_stdext_resources)
```

Comment:
Can we change the name? `xapi_stdext_` is used right only in https://github.com/xapi-project/stdext, making this package be in an unexpected place. I would prefer a name like `managed_resources`, `scoped_resources`, `safe_resources` or even plain `resources`.

Reply:
I renamed it to Safe_resources. Originally my plan was to move this to stdext, but that requires tagging a new xs-opam, which adds additional complexity in merging.